stock_request_analytic/models/stock_request.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2017-2020 ForgeFlow, S.L. (https://www.forgeflow.com)
+# Copyright 2021 Tecnativa - João Marques
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class StockRequest(models.Model):
+    _inherit = "stock.request"
+    _check_company_auto = True
+
+    analytic_account_id = fields.Many2one(
+        "account.analytic.account",
+        string="Analytic Account",
+        check_company=True,
+    )
+    analytic_tag_ids = fields.Many2many(
+        "account.analytic.tag",
+        string="Analytic Tags",
+        check_company=True,
+    )
+
+    @api.onchange("product_id")
+    def onchange_product_id(self):
```

Comment:
This will be removed in further version of Odoo.

To replace it, change the analytic_account_id field to computed, store=True, readonly=False. And change the onchange to compute method.